scene/surface-placement/src/main/java/com/esri/samples/surface_placement/SurfacePlacementController.java

```diff
@@ -0,0 +1,203 @@
+
+      TextSymbol relativeText =
+          new TextSymbol(14, "RELATIVE", 0xFF0000FF, TextSymbol.HorizontalAlignment.LEFT, TextSymbol.VerticalAlignment.MIDDLE);
+      relativeText.setOffsetX(20);
+
+      TextSymbol absoluteText =
+          new TextSymbol(14, "ABSOLUTE", 0xFF0000FF, TextSymbol.HorizontalAlignment.LEFT, TextSymbol.VerticalAlignment.MIDDLE);
+      absoluteText.setOffsetX(20);
+
+      TextSymbol relativeToSceneText =
+        new TextSymbol(14, "RELATIVE TO SCENE", 0xFF0000FF, TextSymbol.HorizontalAlignment.RIGHT, TextSymbol.VerticalAlignment.MIDDLE);
+      relativeToSceneText.setOffsetX(-20);
+
+      // add the point graphic and text graphic to the corresponding graphics overlay
+      drapedBillboardedOverlay.getGraphics().add(new Graphic(surfaceRelatedPoint, triangleSymbol));
+      drapedBillboardedOverlay.getGraphics().add(new Graphic(surfaceRelatedPoint, drapedBillboardedText));
+
+      drapedFlatOverlay.getGraphics().add(new Graphic(surfaceRelatedPoint, triangleSymbol));
+      drapedFlatOverlay.getGraphics().add(new Graphic(surfaceRelatedPoint, drapedFlatText));
+
+      relativeOverlay.getGraphics().add(new Graphic(surfaceRelatedPoint, triangleSymbol));
+      relativeOverlay.getGraphics().add(new Graphic(surfaceRelatedPoint, relativeText));
+
+      absoluteOverlay.getGraphics().add(new Graphic(surfaceRelatedPoint, triangleSymbol));
+      absoluteOverlay.getGraphics().add(new Graphic(surfaceRelatedPoint, absoluteText));
+
+      relativeToSceneOverlay.getGraphics().add(new Graphic(sceneRelatedPoint, triangleSymbol));
+      relativeToSceneOverlay.getGraphics().add(new Graphic(sceneRelatedPoint, relativeToSceneText));
+
+      // add graphics overlays to the scene view
+      sceneView.getGraphicsOverlays().addAll(Arrays.asList(drapedBillboardedOverlay, relativeOverlay, absoluteOverlay, relativeToSceneOverlay));
+
+      // link up the ToggleGroup RadioButtons to the corresponding draped overlay
+      drapedBillboardedRadioButton.setUserData(drapedBillboardedOverlay);
+      drapedFlatRadioButton.setUserData(drapedFlatOverlay);
+
+      // listener for toggles between billboarded and flat draped surface placement modes on the JavaFX ToggleGroup
+      toggleGroup.selectedToggleProperty().addListener((observableValue, oldToggle, newToggle) -> {
+        if (toggleGroup.getSelectedToggle() != null) {
+          sceneView.getGraphicsOverlays().remove(oldToggle.getUserData());
+          sceneView.getGraphicsOverlays().add((GraphicsOverlay) toggleGroup.getSelectedToggle().getUserData());
+        }
+      });
+
+    } catch (Exception e) {
+      // on any exception, print the stack trace
+      e.printStackTrace();
+    }
+
+  }
+
+  /**
+   * Sets the Z-Value to the value selected by the JavaFX slider.
+   */
+
+  @FXML
+  private void changeZValue(){
+        // get z-value from slider
+        double zValueFromSlider = zValueSlider.getValue();
+
+        // create new points with updated z-value
```

Comment:
As above, no harm in adding an article, I think:

```suggestion
        // create new points with the updated z-value
```